src/visualizers/FactorFence.ts

```diff
@@ -0,0 +1,738 @@
+                        this.mouseIndex = myIndex
+                        this.mouseOn = true
+                    }
+
+                    // move upward in preparation for next bar
+                    cumulHt += recHeight
+                }
+            }
+        }
+    }
+
+    resetLoop() {
+        this.frame = 0
+        this.sketch.loop()
+    }
+
+    keyPressed() {
+        this.resetLoop()
+    }
+
+    mouseMoved() {
+        this.resetLoop()
+    }
+
+    mouseWheel(event: MouseEvent) {
+        let scaleFac = 1
+        if (event instanceof WheelEvent) {
+            if (event.deltaY > 0) scaleFac = 1.03
+            else scaleFac = 0.97
+        }
+        this.scaleFactor *= scaleFac
+        this.graphCorner.y = this.graphCorner.y / scaleFac
+        this.resetLoop()
+    }
+
+    // set highlight prime by click
+    mouseClicked() {
+        if (this.mouseOn) {
+            this.highlight = this.mousePrime
+        } else {
+            this.highlight = 0n
+        }
+        this.refreshParams()
+        this.resetLoop()
+    }
+
+    bottomText() {
+        // text size and position
+        this.sketch.textSize(this.textSize / this.scaleFactor)
+        const textPosition = this.textCorner.copy()
+        textPosition.mult(1 / this.scaleFactor)
+
+        // spacing between lines
+        const textIntervalVec = this.sketch.createVector(0, this.textInterval)
+        textIntervalVec.mult(1 / this.scaleFactor)
+
+        // always visible static text info, line by line
+        const info = [
+            'Click select; arrow keys to move; U/O stretch; Y/H raise/lower',
+            'Highlighted prime: ' + this.highlight.toString(),
+        ]
+
+        // colours match graph colours
+        const infoColors = [
+            this.palette.gradientBar.bottom,
+            this.palette.gradientHighlight.bottom,
+        ]
+
+        // display static info line by line
+        for (let i = 0; i < info.length; i++) {
+            this.sketch.fill(infoColors[i])
+            this.sketch.text(info[i], textPosition.x, textPosition.y)
+            textPosition.add(textIntervalVec)
+        }
+
+        // factorization text shown upon mouseover of graph
+        if (this.mouseOn) {
+            // parse out the factorization, categorizing
+            // into primes (a) smaller than the highlighted one,
+            // (b) highlighted one, and (c) bigger ones
+            const factorizationPrimes = this.factorizations[
+                this.mouseIndex
+            ].map(factor => factor.prime)
+            const factorizationPrimesPre = factorizationPrimes.filter(
+                factor => factor < this.mousePrime
+            )
+            const factorizationPrimesMouse = factorizationPrimes.filter(
+                factor => factor == this.mousePrime
+            )
+            const factorizationPrimesPost = factorizationPrimes.filter(
+                factor => factor > this.mousePrime
+            )
+
```

Comment:
You could also use Array indexOf() and lastIndexOf and then take three slices if you wanted to avoid filtering the whole array three times.

Reply:
Question:  this is a trade-off, since that code (if I understand your suggestion) would assume the factors are listed in increasing order, while this code does not.  So it's less robust in some way.  Is that a good tradeoff?

Reply:
Good point. It never occurred to me they wouldn't be increasing. Filtering thrice just felt so redundant. But I won't stress about this. Feel free to make a change or not, and just go ahead and resolve this when you are content. My comments, especially this one, are just suggestions. If you want to use indexOf, you could check the increasing condition when you make factorizationPrimes in the first place.

Reply:
Ok, I thought about this and decided to use the indexing.  Because there are already other places in the code where it is assumed the factors are in order.  This relies on the backend returning the factors in order.  Committed.

Reply:
Sounds fine to me. If you want to be extra careful about this, you could put code to check the factor order in the spot where you build the factorizationPrimes list. Or perhaps even better, just file an issue in backscope noting that primes must be in ascending order in factorizations, a fact that should (a) be added to the backscope documentation, and (b) explicitly tested in one of the backscope automatic tests.

Reply:
I added an issue to the backscope.